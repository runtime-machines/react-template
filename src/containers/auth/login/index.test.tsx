import * as React from 'react';
import { shallow } from 'enzyme';

import { Login } from '.';

jest.mock('react-redux', () => ({
	connect: (): jest.Mock => jest.fn(),
	useDispatch: (): jest.Mock => jest.fn()
}));

jest.mock('@utilities', () => ({
	useAppSelector: jest.fn(),
	Routes: {
		BASE: '/',
		ABOUT: '/about',
		LOGIN: '/login',
		SIGNUP: '/signup',
		SETTINGS: '/settings',
		PASSWORD_RESET: '/reset-password'
	}
}));
describe('Login component', () => {
	it('should render successfully', () => {
		const tree = shallow(<Login />);

		expect(tree).toMatchSnapshot();
	});
});
