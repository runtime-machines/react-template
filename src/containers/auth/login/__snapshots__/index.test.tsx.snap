// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component should render successfully 1`] = `
<Wrapper
  className="o-wrapper--fancy"
>
  <LoginForm
    onSubmit={[Function]}
  >
    <p
      className="c-form__hint"
    >
      Don't have an account?
       
      <Link
        to="/signup"
      >
        Sign up
      </Link>
    </p>
    <p
      className="c-form__hint"
    >
      Forgot password?
       
      <Link
        to="/reset-password"
      >
        Reset password
      </Link>
    </p>
  </LoginForm>
</Wrapper>
`;
